src/Avalonia.Visuals/Media/TextTrimming.cs

```diff
@@ -1,23 +1,71 @@
-﻿namespace Avalonia.Media
+﻿using System;
+using Avalonia.Media.TextFormatting;
+
+namespace Avalonia.Media
 {
     /// <summary>
     /// Describes how text is trimmed when it overflows.
     /// </summary>
-    public enum TextTrimming
+    public abstract class TextTrimming
     {
+        public static char s_defaultEllipsisChar = '\u2026';
+
         /// <summary>
         /// Text is not trimmed.
         /// </summary>
-        None,
+        public static TextTrimming None { get; } = new TextNoneTrimming();
 
         /// <summary>
         /// Text is trimmed at a character boundary. An ellipsis (...) is drawn in place of remaining text.
         /// </summary>
-        CharacterEllipsis,
+        public static TextTrimming CharacterEllipsis { get; } = new TextTrailingTrimming(s_defaultEllipsisChar, false);
 
         /// <summary>
         /// Text is trimmed at a word boundary. An ellipsis (...) is drawn in place of remaining text.
         /// </summary>
-        WordEllipsis
+        public static TextTrimming WordEllipsis { get; } = new TextTrailingTrimming(s_defaultEllipsisChar, true);
+
+        /// <summary>
+        /// Text is trimmed after a given prefix length. An ellipsis (...) is drawn in between prefix and suffix and represents remaining text.
+        /// </summary>
+        public static TextTrimming PrefixCharacterEllipsis { get; } = new TextLeadingPrefixTrimming(s_defaultEllipsisChar, 8);
```

Comment:
Why default to 8? Zero seems to make more sense and behaves the inverse of CharacterEllipsis.

Reply:
Mostly based on AfterEffects default value (which seems to be 7 or 8) since my implementation idea was from there.
0 works as well although from my perspective it should be called `LeadingCharacterEllipsis` then since with 0 there is no prefix.

Reply:
We might change things around when working on css like syntax support for this. Since it would fix problems with trimming permutations and perhaps require a few changes to exposed members. Current API will get refined a bit.

Reply:
Ok, good to know there is past precedent then. It's also a good idea to add another "value" for LeadingCharacterEllipsis. That would allow both cases in XAML. Since length cannot be configured in XAML that might actually be better overall.